src/utils/config.ts

```diff
@@ -47,14 +49,47 @@ export const ContactPersonParams = t.intersection([
 ]);
 export type ContactPersonParams = t.TypeOf<typeof ContactPersonParams>;
 
-const SpidLogsParams = t.union([
-  t.interface({
-    ENABLE_SPID_ACCESS_LOGS: t.literal(true),
+const SpidLogsStorageAzureStorage = t.interface({
+  SPID_LOGS_STORAGE_CONNECTION_STRING: NonEmptyString,
+  SPID_LOGS_STORAGE_CONTAINER_NAME: NonEmptyString,
+  SPID_LOGS_STORAGE_KIND: withDefault(
+    t.literal("azurestorage"),
+    // for backward compatibility, we assume no kind means azure storage
+    "azurestorage"
+  )
+});
 
-    SPID_LOGS_PUBLIC_KEY: NonEmptyString,
-    SPID_LOGS_STORAGE_CONNECTION_STRING: NonEmptyString,
-    SPID_LOGS_STORAGE_CONTAINER_NAME: NonEmptyString
+const SpidLogsStorageAwsS3 = t.intersection([
+  t.interface({
+    AWS_ACCESS_KEY_ID: NonEmptyString,
+    AWS_SECRET_ACCESS_KEY: NonEmptyString,
+    SPID_LOGS_STORAGE_CONTAINER_HOST: NonEmptyString,
+    SPID_LOGS_STORAGE_CONTAINER_NAME: NonEmptyString,
+    SPID_LOGS_STORAGE_CONTAINER_PORT: NonEmptyString,
+    SPID_LOGS_STORAGE_CONTAINER_PROTOCOL: NonEmptyString,
```

Comment:
Why do we need all these variables? Endpoint's parts are already available in `SPID_LOGS_STORAGE_ENDPOINT`.

Also, endpoint should not be mandatory as we don't usually need to specify it.

Thought?

Reply:
As far as I'm concerned, having only `SPID_LOGS_STORAGE_ENDPOINT` as string is enough... and it is not mandatory